From 0c4974d68804cdaff2efb6317f7853a89d3a3d2b Mon Sep 17 00:00:00 2001
From: Antonio Rojas <arojas@archlinux.org>
Date: Mon, 6 Jan 2020 20:30:59 +0100
Subject: [PATCH] Fix build with gpsd 3.20

gps_data_t.online has changed type from double to struct timespec

BUG:425939
Differential Revision: https://phabricator.kde.org/D26474
---
 dataengines/geolocation/location_gps.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dataengines/geolocation/location_gps.cpp b/dataengines/geolocation/location_gps.cpp
index f06865e28..e1a11c6f4 100644
--- a/dataengines/geolocation/location_gps.cpp
+++ b/dataengines/geolocation/location_gps.cpp
@@ -59,7 +59,11 @@ void Gpsd::run()
         if (gps_poll(m_gpsdata) != -1) {
 #endif
             //qDebug() << "poll ok";
+#if GPSD_API_MAJOR_VERSION >= 9
+            if (m_gpsdata->online.tv_sec || m_gpsdata->online.tv_nsec) {
+#else
             if (m_gpsdata->online) {
+#endif
                 //qDebug() << "online";
                 if (m_gpsdata->status != STATUS_NO_FIX) {
                     //qDebug() << "fix";
